community/kernel/src/test/java/org/neo4j/test/TestGraphDatabaseRule.java

```diff
@@ -0,0 +1,297 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+
+import org.neo4j.CommunityTestGraphDatabase;
+import org.neo4j.TestGraphDatabase;
+import org.neo4j.function.Consumer;
+import org.neo4j.function.Consumers;
+import org.neo4j.function.Function;
+import org.neo4j.function.Functions;
+import org.neo4j.function.Supplier;
+import org.neo4j.graphdb.GraphDatabaseService;
+import org.neo4j.graphdb.Label;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.QueryExecutionException;
+import org.neo4j.graphdb.Result;
+import org.neo4j.graphdb.Transaction;
+import org.neo4j.graphdb.mockfs.EphemeralFileSystemAbstraction;
+import org.neo4j.graphdb.schema.Schema;
+import org.neo4j.helpers.Pair;
+import org.neo4j.io.fs.FileUtils;
+import org.neo4j.logging.LogProvider;
+
+public class TestGraphDatabaseRule extends ExternalResource implements Supplier<TestGraphDatabase>
```

Comment:
Wouldn't it be neat if `TestGraphDatabaseRule` implemented `GraphDatabase` directly so that tests wouldn't have to get the db from it and tuck it away in a variable, where the test ends up having two db variables, where one is named `...Rule`. I'd love to get away from that. Of course that could be another PR after this one, but it's just that this PR already changes all these test classes, so why not?